lib/async_hooks.js

```diff
@@ -345,15 +346,10 @@ function emitInitS(asyncId, type, triggerAsyncId, resource) {
 
 function emitBeforeN(asyncId) {
   processing_hook = true;
-  // Use a single try/catch for all hook to avoid setting up one per iteration.
-  try {
-    for (var i = 0; i < active_hooks_array.length; i++) {
-      if (typeof active_hooks_array[i][before_symbol] === 'function') {
-        active_hooks_array[i][before_symbol](asyncId);
-      }
+  for (var i = 0; i < active_hooks_array.length; i++) {
+    if (typeof active_hooks_array[i][before_symbol] === 'function') {
+      active_hooks_array[i][before_symbol](asyncId);
```

Comment:
BTW If you're touching this you could use `for...of` (supposed to be tiny bit faster on TF&I)
Ref: https://github.com/nodejs/node/pull/13419#discussion_r119963668